IOMC/EventVertexGenerators/src/HLLHCEvtVtxGenerator.cc

```diff
@@ -48,35 +47,77 @@ void HLLHCEvtVtxGenerator::fillDescriptions(edm::ConfigurationDescriptions& desc
+    fMeanY = p.getParameter<double>("MeanYIncm") * CLHEP::cm;
+    fMeanZ = p.getParameter<double>("MeanZIncm") * CLHEP::cm;
+    fTimeOffset = p.getParameter<double>("TimeOffsetInns") * CLHEP::ns * c_light;
+    fEProton = p.getParameter<double>("EprotonInGeV") * 1e9;
+    fCrossingAngle = p.getParameter<double>("CrossingAngleInurad") * 1e-6;
+    fCrabFrequency = p.getParameter<double>("CrabFrequencyInMHz") * 1e6;
+    fRF800 = p.getParameter<bool>("RF800");
+    fBetaCrossingPlane = p.getParameter<double>("BetaCrossingPlaneInm");
+    fBetaSeparationPlane = p.getParameter<double>("BetaSeparationPlaneInm");
+    fHorizontalEmittance = p.getParameter<double>("HorizontalEmittance");
+    fVerticalEmittance = p.getParameter<double>("VerticalEmittance");
+    fBunchLength = p.getParameter<double>("BunchLengthInm");
+    fCrabbingAngleCrossing = p.getParameter<double>("CrabbingAngleCrossingInurad") * 1e-6;
+    fCrabbingAngleSeparation = p.getParameter<double>("CrabbingAngleSeparationInurad") * 1e-6;
+    // Set parameters inferred from configurables
+    gamma = fEProton / pmass + 1.0;
+    beta = std::sqrt((1.0 - 1.0 / gamma) * ((1.0 + 1.0 / gamma)));
+    betagamma = beta * gamma;
+    oncc = fCrabbingAngleCrossing / fCrossingAngle;
+    epsx = fHorizontalEmittance / (betagamma);
+    epss = epsx;
+    sigx = std::sqrt(epsx * fBetaCrossingPlane);
+    phiCR = oncc * fCrossingAngle;
+  }
+  if (readDB_) {
+    // NOTE: this is currently watching LS transitions, while it should watch Run transitions,
+    // even though in reality there is no Run Dependent MC (yet) in CMS
+    beamToken_ =
+        esConsumes<SimBeamSpotHLLHCObjects, SimBeamSpotHLLHCObjectsRcd, edm::Transition::BeginLuminosityBlock>();
+  }
+}
+
+void HLLHCEvtVtxGenerator::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const& iEventSetup) {
+  update(iEventSetup);
+}
+
+void HLLHCEvtVtxGenerator::update(const edm::EventSetup& iEventSetup) {
+  if (readDB_ && parameterWatcher_.check(iEventSetup)) {
+    edm::ESHandle<SimBeamSpotHLLHCObjects> beamhandle = iEventSetup.getHandle(beamToken_);
+    // Read configurable parameters
+    fMeanX = beamhandle->meanX() * CLHEP::cm;
+    fMeanY = beamhandle->meanY() * CLHEP::cm;
+    fMeanZ = beamhandle->meanZ() * CLHEP::cm;
+    fEProton = beamhandle->eProton() * 1e9;
+    fCrossingAngle = beamhandle->crossingAngle() * 1e-6;
+    fCrabFrequency = beamhandle->crabFrequency() * 1e6;
+    fRF800 = beamhandle->rf800();
+    fBetaCrossingPlane = beamhandle->betaCrossingPlane();
+    fBetaSeparationPlane = beamhandle->betaSeparationPlane();
+    fHorizontalEmittance = beamhandle->horizontalEmittance();
+    fVerticalEmittance = beamhandle->verticalEmittance();
+    fBunchLength = beamhandle->bunchLenght();
+    fCrabbingAngleCrossing = beamhandle->crabbingAngleCrossing() * 1e-6;
+    fCrabbingAngleSeparation = beamhandle->crabbingAngleSeparation() * 1e-6;
+    fTimeOffset = beamhandle->timeOffset() * CLHEP::ns * c_light;
```

Comment:
@francescobrivio , is fTimeOffset measured in ns or in cm? what are units of c_light? If it is coming from CLHEP then CLHEP:: namespace should be used.

Reply:
Given the parameter name in the cfi, I believe it's in ns.
`fTimeOffset` used to be:
```
const double fTimeOffset;
fTimeOffset(p.getParameter<double>("TimeOffsetInns") * CLHEP::ns * c_light),
```
and now it's:
```
double fTimeOffset;
fTimeOffset = p.getParameter<double>("TimeOffsetInns") * CLHEP::ns * c_light;
```

Yes I think c_light comes from CLHEP...I don't see any other `#include`s that could import it...I'll add the CLHEP:: namespace.

Reply:
@francescobrivio , I guess CLHEP include file is available, because of CLHEP::ns. So, fTimeOffset is defined in mm. WHy the name of this variable fTimeOffset ? If it is really the time, why it is multiplied by c_light?

Reply:
I'm not the original author but this is used only in:
https://github.com/cms-sw/cmssw/blob/581553478a6959f428a3c99f83aa9099e6e0fb82/IOMC/EventVertexGenerators/src/HLLHCEvtVtxGenerator.cc#L81-L118

And it seems it should correctly be transformed to a "lenght"...so I'll simply add the CLHEP:: namespace without actually changing the current behavior.

Reply:
done in d4a9b65c7490af18d27ee8875090d46474455396

Reply:
Yes, this is correct.